crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1419,6 +1420,360 @@ impl<'w, 'a, T: Component> VacantEntry<'w, 'a, T> {
     }
 }
 
+/// Provides read-only access to a single entity and some of its components defined by the contained [`Access`].
+pub struct FilteredEntityRef<'w> {
+    entity: UnsafeEntityCell<'w>,
+    access: Access<ComponentId>,
+}
+
+impl<'w> FilteredEntityRef<'w> {
+    /// # Safety
+    /// - `cell` must have permission to mutate every component of the entity.
+    /// - No accesses to any of the entity's components may exist
+    ///   at the same time as the returned [`FilteredEntityMut`].
```

Comment:
```suggestion
    /// - No `&mut World` can exist from the underlying `UnsafeWorldCell`
    /// - If `access` takes a read on a component, no mutable references to that component
    /// may exist at the same time as the returned [`FilteredEntityRef`].
```

Is this more what the safety requirements should be here?